lib/puppet/provider/oneview_volume/c7000.rb

```diff
@@ -14,33 +14,15 @@
 # limitations under the License.
 ################################################################################
 
-require_relative '../login'
-require_relative '../common'
+require_relative '../oneview_resource'
 require 'oneview-sdk'
 
-Puppet::Type.type(:oneview_volume).provide(:oneview_volume) do
-  mk_resource_methods
+Puppet::Type::Oneview_volume.provide :c7000, parent: Puppet::OneviewResource do
+  desc 'Provider for OneView Storage Volumes using the C7000 variant of the OneView API'
 
-  def initialize(*args)
-    super(*args)
-    @client = OneviewSDK::Client.new(login)
-    @resourcetype = OneviewSDK::Volume
-    # Initializes the data so it is parsed only on exists and accessible throughout the methods
-    # This is not set here due to the 'resources' variable not being accessible in initialize
-    @data = {}
-  end
+  mk_resource_methods
 
-  def self.instances
-    @client = OneviewSDK::Client.new(login)
-    matches = OneviewSDK::Volume.find_by(@client, {})
-    matches.collect do |line|
-      name = line['name']
-      data = line.inspect
-      new(name: name,
-          ensure: :present,
-          data: data)
-    end
-  end
+  confine true: login[:hardware_variant] == 'C7000'
 
   # Provider methods
   def exists?
```

Comment:
This can be reduced using super([:found, :get_attachable_volumes, :get_extra_managed_volume_paths])